base_business_document_import/models/business_document_import.py

```diff
@@ -0,0 +1,1391 @@
+# Copyright 2015-2021 Akretion France
+# @author: Alexis de Lattre <alexis.delattre@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+from urllib.parse import urlparse
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+from odoo.osv import expression
+from odoo.tools import float_compare
+
+from odoo.addons.base_iban.models.res_partner_bank import validate_iban
+
+logger = logging.getLogger(__name__)
+
+
+class BusinessDocumentImport(models.AbstractModel):
+    _name = "business.document.import"
+    _description = "Common methods to import business documents"
+
+    @api.model
+    def user_error_wrap(self, method, data_dict, error_msg):
+        """The method and data_dict arguments are useful when you want to
+        inherit this method to update the error messag_match_currencye"""
+        assert error_msg
+        raise UserError(error_msg)
+
+    def _direct_match(self, data_dict, model, raise_exception=True):
+        if data_dict.get("recordset"):
+            record = data_dict["recordset"]
+            if isinstance(record, type(model)):
+                return record
+            elif raise_exception:
+                raise UserError(
+                    _(
+                        "The record '{record}' is an instance of '{record_model}', "
+                        "not of '{target_model}'."
+                    ).format(
+                        record=record.display_name,
+                        record_model=record._name,
+                        target_model=model._name,
+                    )
```

Comment:
```suggestion
                    this.env._(
                        "The record '%(record)s' is an instance of '%(record_model)s', "
                        "not of '%(target_model)s'.",
                        record=record.display_name,
                        record_model=record._name,
                        target_model=model._name,
                    )
```

Two things here (and everywhere else):
- Use of `this.env._`
- Use the builting interpolation mechanism which lets odoo recover gracefully from bad translation texts

Reply:
:+1: 
But it's actually `self.env._`, you're doing too much JS :stuck_out_tongue_winking_eye:

Reply:
😆 you got me